### PR TITLE
add comment bot to CI

### PR DESCRIPTION
**Pull request by g1y5x3**
Since this PR branch is not the target the of pull request, the test won't be triggered here. Once it is merged to the master branch (or can be any other branches), other PRs targeting the master will trigger this test properly. 

Tested in the following scenarios:
1. created a pr within the fork but different branch using the same account - https://github.com/g1y5x3/tinygrad/pull/5#issuecomment-1722142188
2. created a pr from a different fork and branch but still using the same account - https://github.com/g1y5x3/tinygrad/pull/6#issuecomment-1722145324
3. created a pr from a different fork and branch using a difference account - https://github.com/g1y5x3/tinygrad/pull/7#issuecomment-1722147996

### CODE REVIEW

File: .github/workflows/szdiff.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.sha }}
+          path: pr
+        # the base must be from tinygrad master for security purpose
+      - name: Checkout code from tinygrad master
+        uses: actions/checkout@v4
+        with:
+          repository: tinygrad/tinygrad
+          ref: master
+          path: base
```
Comment (wozeparrot) [inline .github/workflows/szdiff.yml:30]: doesn't actions/checkout default to the base branch when its `pull_request_target`?

Comment (g1y5x3) [reply .github/workflows/szdiff.yml]: Just did a quick test, you are correct.
image

**Comment by wozeparrot** [comment]
try #2 :)

**Comment by g1y5x3** [comment]
It works! https://github.com/tinygrad/tinygrad/pull/1874#issuecomment-1722265032

**Pull request merged**